src/appleseed/renderer/kernel/rendering/pixelrendererbase.cpp

```diff
@@ -99,6 +157,20 @@ void PixelRendererBase::on_pixel_end(
         {
             RENDERER_LOG_WARNING("more invalid samples found, omitting warning messages for brevity.");
         }
+
+    }
+
+    // Invalid samples diagnostic
+    if (tile_bbox.contains(pt) && m_params.m_diagnostics)
```

Comment:
It would make sense to swap the conditions to avoid calling `AABB::contains()` if `m_params.m_diagnostics` is false.